### ISSUE TITLE
identify user language on static pages

### ISSUE THREAD

**Issue by sujato** [suttacentral/suttacentral#2352]
Currently the static pages have the same URL no matter what the user language is.

Instead, we should identify the page per language to enable sharing, eg:

```
https://suttacentral.net/abhidhamma-guide-sujato?lang=en
https://suttacentral.net/abhidhamma-guide-sujato?lang=de
```

**Comment by sabbamitta** [comment]
Thank you! The German Abhidhamma association will be grateful. :smiley:

**Comment by thesunshade** [comment]
I'd like to request that this be re-thought.

Here is the problem. User A has their interface language set to English. The `lang` gets automatically added to all of their urls. 

User A shares the link online. `https://suttacentral.net/abhidhamma-guide-sujato?lang=en`

User B Speaks German and has their site language set to German. Why would they want to be reading an English version of the page when a German one is available?

If localization is working properly (and I think we are getting closer to this) people should be able to choose their language (either manually or by the site automatically detecting language through the browser settings). 

I think it woudl be *ok* if the `lang` parameter was some kind of hack where users in the know could add it. But that parameter should not be added to all url's automatically. 

@sabbamitta I know you like to be sharing links directly to the German translations, but can you see why for general use people should be seeing the page in their chosen language? 

Bhante @sujato is there something I'm missing on this?

Really, I don't think that the `lang` param makes any sense at all in the url. Unless it would be for the hack I mention above, I think it serves no point and clutters things up.

**Comment by thesunshade** [comment]
#### User Story:
[in a well functioning localization universe]

Ven. Sumitta teaches in French and he would like his students to read the Abhidhamma article on SuttaCentral. So he shares the link `suttacentral.net/abhidhamma-guide-sujato` and reminds his French speaking students that their experience of SuttaCentral will be greatly improved if they change their site language to French. Namely, that they can read these articles in French, the interface will be in French, and French translations will be displayed as a priority.

 Fortunately most of his students are already seeing the articles and interface in French because SC auto-detects the prefered language the first time it is opend. 

When Luca, one of his students, hears this, he decideds that his German is better than his French, so he sets his site language to German. From now on any links to static pages that Ven. Sumitta shares will display in German and easier for him to read.

**Comment by sabbamitta** [comment]
>@sabbamitta I know you like to be sharing links directly to the German translations, but can you see why for general use people should be seeing the page in their chosen language?

Yes, I was actually glad when the `?lang=de` became part or the URL because I *thought* this would give me the possibility to share suttas or static pages including language setting. Because I still hear of people that they don't know that SC has a German interface, or language selection at all, so I thought this is a way to make this more widely known.

Until … yes, until recently I tested this in an incognito browser! The result was a bit disappointing because although I had `?lang=de` included most of the site was still shown in English (see screenshots on Gitter).

So. Taking all this into consideration, the first priority should be to get the language setting to work properly. As Ven. @thesunshade calls it, "a well functioning localization universe".

Thanks for picking this up, Bhante!

**Comment by thesunshade** [comment]
> Because I still hear of people that they don't know that SC has a German interface, or language selection at all, so I thought this is a way to make this more widely known.

I completely agree that there is not enough awareness of the internationalization feature. 

However this needs to be solved through technical improvements (such as https://github.com/suttacentral/suttacentral/issues/2731 and using browser language) and education. If someone doesn't know something, then you need to *tell them*, not hijack their their interface.

> Until … yes, until recently I tested this in an incognito browser! The result was a bit disappointing because although I had ?lang=de included most of the site was still shown in English (see screenshots on Gitter).

The major problem with this is the fact that ***every*** link someone shares includes their personal language setting.

If there really is a need to force someone to look at a specific translation of a static page, then there should be some kind of a permalink button. However, I don't believe (outside of administrative work) there is ever a case for using it.
